### PR TITLE
Feature/config prod webpack

### PR DESCRIPTION
**Pull request by HuyAms**
- feat: add mini-css-extract-plugin for prod webpack
- feat: add html webpack plugin for prod webpack
- refactor: remove semicolon and trailing comma
- feat: add clean webpack plugin in prod webpack
- feat: add devtool source-map to prod webpack
- feat: add OptimizeCssAssetsPlugin to webpack prod

### CODE REVIEW

File: webpack/webpack.prod.js [hunk 1]
```diff
@@ -1,17 +1,62 @@
...
   mode: 'production',
 
+  devtool: 'source-map',
+
   output: {
     filename: '[name].[chunkhash].js',
     chunkFilename: '[name].[chunkhash].chunk.js',
-    publicPath: '/',
+    publicPath: '/'
   },
```
Comment (vinhlee95) [inline webpack/webpack.prod.js]: Add optimization with TerserWebpackPlugin along with OptimizeCSSAssetsPlugin:
```
optimization: {
    minimizer: [
      new TerserJSPlugin({}),
      new OptimizeCSSAssetsPlugin({})
    ]
  },
```

File: webpack/webpack.prod.js [hunk 2]
```diff
@@ -1,17 +1,62 @@
...
+
+  plugins: [
+
+    new CleanWebpackPlugin(),
+
+    new MiniCssExtractPlugin({
+      filename: 'style.[contenthash].css'
+    }),
+
+    new OptimizeCssAssetsPlugin(),
```
Comment (vinhlee95) [inline webpack/webpack.prod.js]: Should probably use in `optimization` above

File: webpack/webpack.prod.js (hunk 1 above)
Comment (vinhlee95) [reply webpack/webpack.prod.js]: I saw people use `splitChunk` in `optimization` as well: 
https://webpack.js.org/plugins/split-chunks-plugin/
Haven't gone thru its real benefits but probably 1 consideration

File: webpack/webpack.prod.js [hunk 3]
```diff
@@ -1,17 +1,62 @@
...
-  }
+  },
+
+  plugins: [
+
+    new CleanWebpackPlugin(),
+
+    new MiniCssExtractPlugin({
+      filename: 'style.[contenthash].css'
+    }),
```
Comment (vinhlee95) [inline webpack/webpack.prod.js]: As we split outputs to chunks, should probably have `chunkFileName` option for this plugin as well:
```
new MiniCssExtractPlugin({
	filename: '[name].[hash].css',
	chunkFilename: '[id].[hash].css'
}),
```

File: webpack/webpack.prod.js (hunk 1 above)
Comment (HuyAms) [reply webpack/webpack.prod.js]: Webpack 4 has mode feature, when set it to production, it has TerserJSPlugin() by default. No need to override optimization to include TerserJSPlugin

https://webpack.js.org/configuration/mode/#mode-production

**Pull request merged**